packages/nimble-components/src/table/specs/table-column-specs/table-column-mapping.md

```diff
@@ -0,0 +1,331 @@
+    html<Mapping, TableColumnMappingHeaderView>`
+        ${when((x, c) => x.key === (c.parent as TableColumnMappingHeaderView).groupHeaderValue,
+            html<Mapping>`${x => x.groupHeaderViewTemplate}`
+        )}
+    `
+)}
+```
+
+#### `nimble-mapping-*`
+
+```HTML
+<template slot="mapping"></template>
+```
+
+#### `nimble-mapping-icon`
+
+Cell view template:
+
+```HTML
+<${this.icon}
+    title="${x => x.label}"
+    aria-label="${x => x.label}"
+    severity="${x => x.severity}">
+</${this.icon}>
+```
+
+Group header view template:
+
+```HTML
+<${this.icon}
+    title="${x => x.label}"
+    aria-label="${x => x.label}"
+    severity="${x => x.severity}">
+</${this.icon}>
+<span
+    ${ref('span')}
+    @mouseover="${x => {
+        x.isValidContentAndHasOverflow = !!x.label && x.span!.offsetWidth < x.span!.scrollWidth;
+    }}"
+    @mouseout="${x => {
+        x.isValidContentAndHasOverflow = false;
+    }}"
+    title=${x => (x.isValidContentAndHasOverflow ? x.label : null)}>
+    ${x => x.label}
+</span>`;
+```
+
+### Grouping
+
+Grouping will be based on the record value. The grouping header will display the rendered icon/spinner/text. In the case of an icon/spinner, it will also be followed by the `label` text. This will disambiguate cases where multiple record values map to the same icon (assuming the labels are different).
+
+For values that do not match any mapping, we will display the raw data value. While this introduces inconsistency, it seems preferable to the alternative, which is having multiple, separate groupings with a blank header (well, with just the item count in parens). Even in the case where there is a default mapping, we would still end up with separate groups with the identical default mapped icon and/or text, which is just a bad.
+
+Text in a grouping header will be ellipsized and gain a tooltip if there is not enough room to display it all.
+
+### Sorting
+
+Sorting will be based on the record value. For boolean and number values, a basic sort (just using basic comparison/equality operators) is the clear choice. For string values, it is less clear. In the case where the strings are enum values, they are likely to be non-localized, English strings. In that case, even if there is a semantically meaningful sort order (i.e. "NOT_STARTED" < "RUNNING" < "DONE"), we can't sort that way. Settling for a basic sort in this case seems just as reasonable as anything else. If the strings are localized, it's more likely the client would use a text column instead. If we did want to support a different sorting method for strings than booleans/numbers, it would be awkward to implement. We would have update the column's sort method when the data changes, and the column would have to determine the data type for its field, which might require looking at multiple records (e.g. if the first didn't define that field). For these reasons, we will just use a basic sort for all supported data types (string, boolean and number).
```

Comment:
It sounds like the concern here is having a clear way for users to express the sorting of the mappings.
That is already expressed one way. We could just use the order of the mapping elements in the DOM.

Reply:
That's an interesting idea. The current implementation only supports specifying the sort function (by an enum value) to one of a couple presets. I don't know if there's a good way to make it support using a function bound to an instance of my column (which I would need to access the ordered mappings). I'll have to think a bit.

Reply:
We were reset but looks like this thread is still open. Do we know if there are current expected use-cases of needing to sort mapped values? Maybe we can discuss this at stand-up?

Reply:
After discussing at stand-up, I've update the spec with the proposed options. I'm suggesting we use a basic sort initially, then add an enumerated sort as a follow-on PR.

Reply:
The spec does not seem to reflect your comment. It seems to just have verbose language trying to justify why any other sort option is bad instead of choosing basic sort today and sorting via dom ordering of mapping elements as future work.

Reply:
This issue is this commit wasn't showing up for me: https://github.com/ni/nimble/pull/1220/commits/7e27c21438950947e7deaf9fc2682fb8d3079b80
Maybe was accidentally not pushed or I needed to refresh or something.
But that commit and the follow-on of https://github.com/ni/nimble/pull/1220/commits/643cd8f5cc4a87fe01f5d91e495958c94e937e98 together look great.